pkg/proxy/healthcheck/healthcheck.go

```diff
@@ -20,108 +20,194 @@ import (
 	"fmt"
 	"net"
 	"net/http"
+	"strings"
+	"sync"
 
 	"github.com/golang/glog"
+	"github.com/renstrom/dedent"
+
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/client-go/tools/cache"
 )
 
-// proxyMutationRequest: Message to request addition/deletion of endpoints for a service
-type proxyMutationRequest struct {
-	serviceName  types.NamespacedName
-	endpointUids *sets.String
+// Server serves HTTP endpoints for each service name, with results
+// based on the endpoints.  If there are 0 endpoints for a service, it returns a
+// 503 "Service Unavailable" error (telling LBs not to use this node).  If there
+// are 1 or more endpoints, it returns a 200 "OK".
+type Server interface {
```

Comment:
I did that originally, and `go lint` failed me.  Scoped names should be succinct.  `healthcheck.Server` vs `healthcheck.HealthCheckServer`